### DISCUSSION TITLE
How to understand the "bond graph"?

### DISCUSSION THREAD

**Discussion by Asecretboy** [CederGroupHub/chgnet#138] [Q&A] [answered]
I am a computer science major, and I don't have a deep understanding of materials science. I find it challenging to comprehend the representation of the "bond graph" in the chgnet code. Why is there a distinction between directed edges and undirected edges in crystal structures? Additionally, how does the following definition effectively represent the bond graph?
            bond graph:
                [[0, 1, 1, 2, 2],
                [0, 1, 1, 4, 23],
                [1, 4, 23, 5, 66],
                ... ...  ]
                the fist column specifies node(atom) index at this angle,
                the second column specifies 1st undirected edge(left bond) index,
                the third column specifies 1st directed edge(left bond) index,
                the fourth column specifies 2nd undirected edge(right bond) index,
                the fifth column specifies 2nd directed edge(right bond) index,.

**Comment by bowen-bd** [comment, marked as answer]
In a crystal: (bond-angle-bond) relation is represemted in the bond-graph by (node-edge-node), where the nodes are undirected bond vectors and edge is the angle embedding + atom embedding. Undirected bond means the bond from atom1 to atom2 share weight with the bond from atom2 to atom1.
